util/src/entities/Application.ts

```diff
@@ -8,60 +8,107 @@ import { User } from "./User";
 export class Application extends BaseClass {
 	@Column()
 	name: string;
-
+	
 	@Column({ nullable: true })
 	icon?: string;
-
-	@Column()
+	
+	@Column({ nullable: true })
 	description: string;
-
-	@Column({ type: "simple-array", nullable: true })
-	rpc_origins?: string[];
-
+	
+	@Column({ nullable: true })
+	summary: string = "";
+	
+	@Column({ type: "simple-json", nullable: true })
+	type?: any;
+	
 	@Column()
-	bot_public: boolean;
-
+	hook: boolean = true;
+	
+	@Column()
+	bot_public?: boolean = true;
+	
+	@Column()
+	bot_require_code_grant?: boolean = false;
+	
 	@Column()
-	bot_require_code_grant: boolean;
+	verify_key: string;
+	
+	@JoinColumn({ name: "owner_id" })
+	@ManyToOne(() => User)
+	owner: User;
+	
+	@Column()
+	flags: number = 0;
+	
+	@Column({ type: "simple-array", nullable: true })
+	redirect_uris: string[] = [];
+	
+	@Column({ nullable: true })
+	rpc_application_state: number = 0;
+	
+	@Column({ nullable: true })
+	store_application_state: number = 1;
+	
+	@Column({ nullable: true })
+	verification_state: number = 1;
+	
+	@Column({ nullable: true })
+	interactions_endpoint_url?: string;
+	
+	@Column({ nullable: true })
+	integration_public: boolean = true;
+	
+	@Column({ nullable: true })
+	integration_require_code_grant: boolean = false;
+	
+	@Column({ nullable: true })
+	discoverability_state: number = 1;
+	
+	@Column({ nullable: true })
+	discovery_eligibility_flags: number = 2240;
```

Comment:
default should be the setting that makes everyone eligible for discovery

Reply:
Not really possible when we don't know what the possible flags and states are

Reply:
while we do not know what bit corresponds to what condition, 0 in each means not required and 1 means required